### ISSUE TITLE
NFS4 volume not mounted after restart docker 1.12.0

### ISSUE THREAD

**Issue by Noodle05** [docker/docker#25310]
Docker version:
```
$ docker version
Client:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 22:11:10 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 22:11:10 2016
 OS/Arch:      linux/amd64
```

Docker info:
```
$ docker info
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 5
Server Version: 1.12.0
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 25
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: host bridge null overlay
Swarm: active
 NodeID: b2127oskqy8yliaw6gouueyri
 Is Manager: false
 Node Address: 192.168.1.35
Runtimes: runc
Default Runtime: runc
Security Options: apparmor seccomp
Kernel Version: 4.4.0-31-generic
Operating System: Ubuntu 16.04.1 LTS
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 1.953 GiB
Name: test-node.example.com
ID: Y22L:OSKO:V3UH:YJO6:WO57:E7QY:FX5A:J6JP:3LZH:OHCQ:PCSD:J4TD
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
Insecure Registries:
 127.0.0.0/8
```

Create NFS4 volume:
```
$ docker volume create -d local -o type=nfs4 -o o=addr=<NFS Server IP>,rw -o device=:/export/data --name foo
```
Check volume:
```
$ docker volume ls
DRIVER              VOLUME NAME
local               foo
```
Inspect volume
```
$ docker volume inspect foo
[
    {
        "Name": "foo",
        "Driver": "local",
        "Mountpoint": "/var/lib/docker/volumes/foo/_data",
        "Labels": {},
        "Scope": "local"
    }
]
```

Check mount (multiple times, all fine)
```
$ docker run --rm -it -v foo:/data alpine:3.4 ls /data
<Get all files>
```
Restart docker
```
$ sudo systemctl restart docker
```
Check volume (still there)
```
$ docker volume ls
DRIVER              VOLUME NAME
local               foo
```
Inspect volume (still same)
```
$ docker volume inspect foo
[
    {
        "Name": "foo",
        "Driver": "local",
        "Mountpoint": "/var/lib/docker/volumes/foo/_data",
        "Labels": {},
        "Scope": "local"
    }
]
```
Check mount to container
```
$ docker run --rm -it -v foo:/data alpine:3.4 ls /data
<No data>
```

**Comment by icecrime** [comment]
Ping @cpuguy83 @tiborvass

**Comment by cpuguy83** [comment]
Fixed in #25316
